### ISSUE TITLE
Unescaped string passed to wget / curl in files.download

### ISSUE THREAD

**Issue by s-vx** [Fizzadar/pyinfra#515]
Hi, 

I just started using pyinfra and so far I like it 🚀, I think I found a tiny little annoyance in the code, so I'm documenting it here hoping that the magic of the Internet ✨ will help someone else in the same situation.   

#### TL;DR:
https://github.com/Fizzadar/pyinfra/blob/6013c813a8eeef63be8c610b76b5dac642b3a7dd/pyinfra/operations/files.py#L116-L117

The string passed as input must be shell escaped (`&`, `;`, etc.) 
 
--- 
I used the following code to install VSCode from a deb url
```python
from pyinfra.operations import apt
url = "https://code.visualstudio.com/sha/download?build=stable&os=linux-deb-x64"
apt.deb(name="vscode", src=url)
```
When I executed the code using wget I've got this error message -> `sh: 1: -O: not found`: 
```python
--> Starting operation: vscode 
    [pyinfra.api.operations] Starting operation vscode on @local
[pyinfra.api.connectors.local] --> Running command on localhost: sh -c 'wget -q https://code.visualstudio.com/sha/download?build=stable&os=linux-deb-x64 -O /tmp/pyinfra-d7b2807d7bab7b216bca33fe176ce152e4bedbf5 || (rm -f /tmp/pyinfra-d7b2807d7bab7b216bca33fe176ce152e4bedbf5; exit 1)'
[pyinfra.api.connectors.util] --> Waiting for exit status...
[pyinfra.api.connectors.util] --> Command exit status: 1
    [@local] sh: 1: -O: not found
    [@local] Error
    [pyinfra.api.state] Failing hosts: @local
```
And a similar error message occured on curl -> `sh: 1: -o: not found`: 

In both cases, the ampersand `&` in the URL was evaluated by the shell (and any other shell constructs could potentially do the same, I've tested with `; ls` at the end of the url and got shell output). A quick and dirty fix was to trick the shell and escape the `&` in the URL: 

```python
url = url.replace("&", "\&") 
```

Other fix can include, but are not limited to: 
* add single quotes to the wget/curl calls `'{0}'` 
* check the src parameter using urllib's urlparse
* let it be 🎶

**Comment by Fizzadar** [comment]
Hi @s-vx, glad to hear you are enjoying using pyinfra :) This is indeed a bug I can't believe I missed this at the time! Fix pushed in https://github.com/Fizzadar/pyinfra/commit/0e34c21a7565b840bd5e1538dc876828e3fac47b which just makes the `1.3.2` cut which I shall release v.shortly!

**Comment by Fizzadar** [comment]
Now released in `v1.3.2`!
